pkg/transport/proxy/transparent/transparent_proxy.go

```diff
@@ -305,10 +305,19 @@ func (p *TransparentProxy) Start(ctx context.Context) error {
 	}
 	p.listener = ln
 
-	// Add auth info endpoint if handler is provided (no middlewares)
+	// Add .well-known path space handler if auth info handler is provided (no middlewares)
 	if p.authInfoHandler != nil {
-		mux.Handle("/.well-known/oauth-protected-resource", p.authInfoHandler)
-		logger.Info("Auth info endpoint enabled at /.well-known/oauth-protected-resource")
+		// Create a handler that routes .well-known requests
+		wellKnownHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
To be honest I forgot the precise steps that led me to adding this but there was a case where the client didn't like the auto-discovery URL, tried falling back to another well-known auto-discovery URL under `/.well-known` and because the whole subtree was protected by middleware, the client received a 401 and just threw an erorr.